pkg/controller/daemon/daemon_controller.go

```diff
@@ -1351,7 +1351,7 @@ func NodeConditionPredicates(nodeInfo *schedulercache.NodeInfo) (bool, []algorit
 		// TODO: There are other node status that the DaemonSet should ideally respect too,
 		//       e.g. MemoryPressure, and DiskPressure
 		if c.Type == v1.NodeOutOfDisk && c.Status == v1.ConditionTrue {
-			reasons = append(reasons, predicates.ErrNodeSelectorNotMatch)
+			reasons = append(reasons, predicates.ErrNodeOutOfDisk)
```

Comment:
Would you add a unit test to make sure the right reasons are returned in DaemonSet predicates? Something similar to:
https://github.com/kubernetes/kubernetes/blob/v1.8.0-alpha.2/plugin/pkg/scheduler/algorithm/predicates/predicates_test.go#L1951

Reply:
Done; I only add one case for this PR; I'll open other PR to increase the coverage.